libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/fakeResolver.kt

```diff
@@ -222,31 +250,25 @@ private fun buildFieldOfNonNullType(
   }
 }
 
-class DefaultFakeResolver(val types: List<CompiledNamedType>) : FakeResolver {
-  private var currentInt = 0
-  private var currentFloat = 0.0
-  private var currentBoolean = false
-
-  private var impl = mutableMapOf<String, Int>()
+open class DefaultFakeResolver(types: List<CompiledNamedType>) : FakeResolver {
+  private val enumTypes = types.filterIsInstance<EnumType>()
+  private val allTypes = types
 
   override fun resolveLeaf(context: FakeResolverContext): Any {
     return when (val name = context.mergedField.type.rawType().name) {
-      "Int" -> currentInt++
-      "Float" -> currentFloat++
-      "Boolean" -> (!currentBoolean).also { currentBoolean = it }
+      "Int" -> context.cacheKey.hashCode() % 100
+      "Float" -> (context.cacheKey.hashCode() % 100000).toFloat() / 100.0
+      "Boolean" -> context.cacheKey.hashCode() % 2 == 0
       "String" -> {
         val index = context.path.indexOfLast { it is String }
-        context.path.subList(index, context.path.size).joinToString { it.toPathComponent() }
+        context.path.subList(index, context.path.size).joinToString(separator = "") { it.toPathComponent() }
```

Comment:
The index only appears in the default string if it's a list of strings right? So this doesn't have the same problem as with ID because a string can't have a cacheKey?

Reply:
The important thing there is `context.path.indexOfLast { it is String }`. It will make the return value depend only on the name of the field. Back to the user case, it means it's going to be `"contactMobile"` or `address[0]` and this doesn't depend what path the user is. It's always going to be `"contactMobile"`. 

It'sa bit naive as an implementation. Ultimately I'd like to provide an implementation based on [kotlin-faker](https://serpro69.github.io/kotlin-faker/) but as far as determinism is concerned it should be good.